### PR TITLE
Add initial SAML documentation

### PR DESCRIPTION
**Pull request by msteffen**

### CODE REVIEW

File: doc/auth/saml_setup.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+1. Activating Pachyderm enterprise and Pachyderm auth
+1. Configuring Pachyderm's auth system to enable its SAML ACS and receive SAML
+    assertions
+1. Logging in to both the dash and CLI
+1. Enabling debug logging in case anything goes wrong
+
+## Activation
+
+For testing, we **highly** recomment running Pachyderm in Minikube, in case any
```
Comment (JoeyZwicker) [inline doc/auth/saml_setup.md]: typo

File: doc/auth/saml_setup.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+   running
+   ```
+   $ pachctl auth use-auth-token
+   Please paste your Pachyderm auth token:
+   <paste robot token emitted by "pachctl auth activate --initial-admin=robot:admin">
+   $ # you are now robot:admin, cluster administrator
+   ```
+
+## Create IdP test app
+Here is the configuration I used for an Okta test app that authenticates Okta users
```
Comment (JoeyZwicker) [inline doc/auth/saml_setup.md]: Don't use "I". This are docs, be prescriptive.

File: doc/auth/saml_setup.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+with Pachyderm:
+![Okta test app config](images/okta_form.png)
+
+Once created, I was able to get the IdP Metadata URL associated with this app here:
```
Comment (JoeyZwicker) [inline doc/auth/saml_setup.md]: same as above.

File: doc/auth/saml_setup.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  ],
+
+  "saml_svc_options": {
+    "acs_url": "http://localhost:30654/saml/acs",
+    "metadata_url": "http://localhost:30654/saml/metadata",
+    "dash_url": "http://localhost:30080/auth/autologin?lol=wut",
+    "session_duration": "8h",
+  }
+}
+EOF
```
Comment (JoeyZwicker) [inline doc/auth/saml_setup.md]: Do we need EOF on all of these?

File: doc/auth/saml_usage.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+023a478b16e849b4996c19632fee6782 montage/e3dd7e9cacc5450c92e0e62ab844bd26 2 minutes ago 8 seconds 0       1 + 0 / 1 371.9KiB 1.283MiB success
+fe8b409e0db54f96bbb757d4d0679186 edges/9cc634a63f794a14a78e931bea47fa73   2 minutes ago 5 seconds 0       2 + 1 / 3 181.1KiB 111.4KiB success
+152cb8a0b0854d44affb4bf4bd57228f montage/82a49260595246fe8f6a7d381e092650 2 minutes ago 5 seconds 0       1 + 0 / 1 79.49KiB 378.6KiB success
+86e6eb4ae1e74745b993c2e47eba05e9 edges/ee7ebdddd31d46d1af10cee25f17870b   2 minutes ago 4 seconds 0       1 + 0 / 1 57.27KiB 22.22KiB success
+```
+
+## Authenticating via a SAML ID Provider (in the dashboard)
+Before authenticating, navigating to the dash will yield a blank screen:
+
+**Blocked-out dash**
```
Comment (JoeyZwicker) [inline doc/auth/saml_usage.md]: Are you going to add images for these?

File: doc/auth/saml_setup.md (hunk 3 above)
Comment (msteffen) [reply doc/auth/saml_setup.md]: Fixed

**Pull request merged**